lib/logging_http_client.rb

```diff
@@ -0,0 +1,33 @@
+require 'time'
+
+class LoggingHttpClient < ::Io::Flow::V0::HttpClient::DefaultHttpHandler
+
+  def execute(request)
+    original_open = client.open_timeout
+    original_read = client.read_timeout
+
+    start_time = Time.now.utc.round(10)
+    puts "[%s] start %s %s" % [start_time.iso8601(6), request.method, request.path]
+
+    if request.path.start_with?("/organizations")
+      # Contrived example to show how client settings can be adjusted
+      client.open_timeout = 60
+      client.read_timeout = 60
+    end
+
+    begin
```

Comment:
Nice example. As soon as I see the remember-stuff-do-something-restore-stuff pattern I start thinking it would be nice to have a method to take care of all of that, but this is after all an example.